src/schema/meta/context.yaml

```diff
@@ -0,0 +1,188 @@
+                subjects:
+                    description: "Collections of subjects in dataset"
+                    type: map
+                    properties:
+                        sub_dirs:
+                            description: "Subjects as determined by sub-*/ directories"
+                            type: seq
+                            items:
+                                type: str
+                        participant_id:
+                            description: "The participant_id column of participants.tsv"
+                            type: seq
+                            items:
+                                type: str
+                        phenotype:
+                            description: "The union of participant_id columns in phenotype files"
+                            type: seq
+                            items:
+                                type: str
+        subject:
+            description: "Properties and contents of the current subject"
+            type: map
+            properties:
+                sessions:
+                    description: "Collections of sessions in subject"
+                    type: map
+                    properties:
+                        ses_dirs:
+                            description: "Sessions as determined by ses-*/ directories"
+                            type: seq
+                            items:
+                                type: str
+                        session_id:
+                            description: "The session_id column of sessions.tsv"
+                            type: seq
+                            items:
+                                type: str
+                        phenotype:
+                            description: "The union of session_id columns in phenotype files"
+                            type: seq
+                            items:
+                                type: str
+
+        # Properties of the current file
+        path:
+            description: "Path of the current file"
+            type: str
+        entities:
+            description: "Entities parsed from the current filename"
+            type: map
+        datatype:
+            description: "Datatype of current file, e.g., anat"
+            type: str
+        modality:
+            description: "Modality of current file, e.g., MRI"
+            type: str
+        sidecar:
+            description: "Sidecar metadata constructed via the inheritance principle"
+            type: map
+        associations:
+            description: |
+                Associated files, indexed by suffix, selected according to the inheritance principle
+            type: map
+            properties:
+                events:
+                    description: "Path to associated events file"
+                    type: str
+                aslcontext:
+                    description: "Path to associated aslcontext file"
+                    type: str
```

Comment:
Do we want to hardcode associations in this file? There are a lot of them, and they all depend on the original file's suffix and/or entities.